src/JSInterop/Microsoft.JSInterop/src/JSObjectReferenceExtensions.cs

```diff
@@ -28,7 +28,7 @@ public static async ValueTask InvokeVoidAsync(this IJSObjectReference jsObjectRe
                 throw new ArgumentNullException(nameof(jsObjectReference));
             }
 
-            await jsObjectReference.InvokeAsync<object>(identifier, args);
+            await jsObjectReference.InvokeAsync<IJSVoidResult>(identifier, args);
```

Comment:
If it's a void type result why do we need to keep it? also what's the difference between InvokeVoidAsync vs InvokeAsync<IJsVoidResult> ??

Reply:
> If it's a void type result why do we need to keep it? 

We still need to have JS return so that we know when to stop awaiting. We could build a separate pipeline which doesn't return any value but that'd add unnecessary complexity / increase long term maintenance. 

> also what's the difference between InvokeVoidAsync vs InvokeAsync ??

`InvokeVoidAsync` is used for calls not requiring a return result, whereas `InvokeAsync<T>` expects a generic `T` result. Note, `InvokeAsync<IJSVoidResult>` should **not** be used in developer code (it's for framework internal use).